app/models/state_file_w2.rb

```diff
@@ -25,13 +25,43 @@ class StateFileW2 < ApplicationRecord
   validates :w2_index, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   validates :employer_state_id_num, format: { with: /\A(\d{0,17})\z/ }
-  validates :state_wages_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :state_income_tax_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :local_wages_and_tips_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :local_income_tax_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :locality_nm, presence: true, if: -> { local_wages_and_tips_amt.present? && local_wages_and_tips_amt.positive? }
-  validates :local_wages_and_tips_amt, numericality: { only_integer: true, greater_than_or_equal_to: 1 }, if: -> { local_income_tax_amt.present? && local_income_tax_amt.positive? }
-  validates :state_wages_amt, numericality: { only_integer: true, greater_than_or_equal_to: 1 }, if: -> { state_income_tax_amt.present? && state_income_tax_amt.positive? }
+  validates :state_wages_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { state_wages_amt.present? }
+  validates :state_income_tax_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { state_income_tax_amt.present? }
+  validates :local_wages_and_tips_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { local_wages_and_tips_amt.present? }
+  validates :local_income_tax_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { local_income_tax_amt.present? }
+  validates :locality_nm, presence: { message: -> (_object, _data) { I18n.t('state_file.questions.ny_w2.edit.locality_nm_missing_error') } }, if: -> { local_wages_and_tips_amt.present? && local_wages_and_tips_amt.positive? }
   validates :employer_state_id_num, presence: true, if: -> { state_wages_amt.present? && state_wages_amt.positive? }
+  validates :locality_nm, format: { with: /\A[a-zA-Z]{1}([A-Za-z\-\s']{0,137})\z/ }, if: -> { locality_nm.present? }
```

Comment:
just to make sure i understand this regex: it's checking that the first character is a letter and then the rest can be either letters or whitespace up to a length of 137?

Reply:
Yes